src/helpers/HelperAssert.sol

```diff
@@ -354,4 +354,112 @@ abstract contract HelperAssert is HelperBase {
         return string(abi.encodePacked("Invalid: ", aStr, operator, bStr, ", reason: ", reason));
     }
 
+    function allowErrors(  
+        bytes memory errorData,
+        string[] memory allowedRequireErrorMessages
+    ) public {
+        allowErrors(errorData, allowedRequireErrorMessages, new bytes4[](0), "", false);
+    }
+    
+    function allowErrors(  
+        bytes memory errorData,
+        bytes4[] memory allowedCustomErrors,
+        string memory errorContext
+    ) public {
+        allowErrors(errorData, new string[](0), allowedCustomErrors, errorContext, false);
+    }
+    
+    function allowErrors(  
+        bytes memory errorData,
+        string[] memory allowedRequireErrorMessages,
+        bytes4[] memory allowedCustomErrors,
+        string memory errorContext
+    ) public {
+        allowErrors(errorData, allowedRequireErrorMessages, allowedCustomErrors, errorContext, false);
+    }
+    
+    /**
+     * Allow require failure & custom errors
+     * @param errorData: return data from a function call. In this case, it's a failure data
+     * @param allowedRequireErrorMessages: allowed require failure messages. It's a string array
+     * @param allowedCustomErrors: allowed custom errors. It can be just 4 bytes function selector or it could be longer
+     * @param errorContext: error context: A message to describe the error
+     * @param allowEmptyError: allow require failure without message
+     */
+    function allowErrors(  
+        bytes memory errorData,
+        string[] memory allowedRequireErrorMessages,
+        bytes4[] memory allowedCustomErrors,
+        string memory errorContext,
+        bool allowEmptyError
+    ) public {
+        if (allowEmptyError && errorData.length == 0) {
+            // 1. empty error case (ex: require(false)) <= useful when there is no error message, errorData is empty
+            // NOTE: Becareful! It could be "require" or "revert" failure without message BUT also it could be something else such as 
+            // calling non-existing address (ex: address(0xdead).call(...)). The root-cause could be various. Please check the 
+            // code and the context before use allowEmptyError = true.
+            _handleEmptyError(allowEmptyError);
+            return;
+        }
+    
+        if (errorData.length < 4) {
+            t(false, "unexpected error data length during allowErrors()");
+            return;
+        }
+    
+        bytes4 selector = bytes4(errorData);
+        
+        if (_isErrorString(selector)) {
+            // 2. require failure case (ex: require(false, "error message"))
+            _allowRequireFailure(errorData, allowedRequireErrorMessages);
+        } else if (allowedCustomErrors.length > 0) {
```

Comment:
not sure i fully understand this if, you are checking to see if there are any allowed custom errors, but you're not checking this for the require-style errors?

also, in the final else, it is considered a unexpected error type, but wouldn't this just mean that the user did not provide any custom error signatures? meaning nothing is allowed i guess